src/core/symbols/html_symbol_report.cpp

```diff
@@ -0,0 +1,285 @@
+		});
+		return QString::fromLatin1(
+		           "<table>\n"
+		           "<tbody>\n"
+		           "%1"
+		           "</tbody>\n"
+		           "</table>\n"
+		           ).arg(
+		        color_data );
+	}
+	
+	QString makeSymbolRow(const Symbol& s)
+	{
+		auto label = QString { s.getNumberAsString()
+		                       + QChar::Space
+		                       + QTextDocumentFragment::fromHtml(s.getName()).toPlainText() };
+		auto extra_text = QString{};
+		if (s.isRotatable())
+			extra_text += QCoreApplication::translate("OpenOrienteering::SymbolReport", "[X] %1")
+			              .arg(QCoreApplication::translate("OpenOrienteering::SymbolReport", "Symbol orientation can be changed."))
+			              + QLatin1String("<br>\n");
+		if (s.hasRotatableFillPattern())
+			extra_text += QCoreApplication::translate("OpenOrienteering::SymbolReport", "[X] %1")
+			              .arg(QCoreApplication::translate("OpenOrienteering::SymbolReport", "Pattern orientation can be changed."))
+			              + QLatin1String("<br>\n");
+		if (s.isHelperSymbol())
+			extra_text += QCoreApplication::translate("OpenOrienteering::SymbolReport", "[X] %1")
+			              .arg(QCoreApplication::translate("OpenOrienteering::SymbolReport", "Helper symbol"))
+			              + QLatin1String("<br>\n");
+		return QString::fromLatin1(
+		           "<tr>"
+		           "<td style=\"vertical-align:top;\">%1</td>\n"          // icon
+		           "<td style=\"vertical-align:middle;\"><b>%2</b></td>"  // label
+		           "</tr>\n"
+		           "<tr>"
+		           "<td>&nbsp;</td>\n"
+		           "<td style=\"padding-bottom:18px;\">\n"
+		           "<div>\n%3</div>\n"  // description
+		           "<p>%4</p>\n"        // configuration properties
+		           "%5</td>"            // colors
+		           "</tr>\n"
+		           ).arg(
+		        imgForSymbol(s, label),
+		        label,
+		        QString(s.getDescription()).replace(QChar::LineFeed, QLatin1String("<br>\n")),
+		        extra_text,
+		        colorsForSymbol(s) );
+	}
+	
+	QString makeSymbolSection()
+	{
+		QString symbol_data;
+		map.applyOnAllSymbols([this, &symbol_data](const Symbol* s){
+			symbol_data.append(makeSymbolRow(*s));
+		});
+		return QString::fromLatin1(
+		           "<h2>%1</h2>\n"
+		           "<table style=\"max-width:60em;\">\n"
+		           "<tbody>\n"
+		           "%2"
+		           "</tbody>\n"
+		           "</table>\n"
+		           ).arg(
+		        QCoreApplication::translate("OpenOrienteering::SymbolReport", "Symbols"),
+		        symbol_data );
+	}
+	
+public:
+	explicit HTMLSymbolReportGenerator(const Map& map)
+	: map(map)
+	{}
+	
+	QString generate()
+	{
+		return QString::fromLatin1(
+		           "<!DOCTYPE html>\n"
+		           "<html>\n"
+		           "<head>\n"
+		           "<meta charset=\"utf-8\">"
+		           "<title>%0</title>\n"
+		           "<meta name=\"generator\" content=\"OpenOrienteering Mapper\">\n"
```

Comment:
I personally prefer some space around the text and a more distinct table header:
```
<style>
table {
  border-spacing: 0px 2px;
}
th {
  text-align: left;
  padding-bottom: 5px;
  font-size: 120%;
}
td,th {
  padding-right: 15px;
}
td:first-child {
  padding-right: 5px;
}
th:first-child {
  padding-right: 5px;
}
</style>
```

Reply:
Actual implementation included in this change:
https://github.com/OpenOrienteering/mapper/compare/3cd842fef149f95c78d4a72251741d2b309fa4ee..870f577e64cbb4a48adac8062cd79bb05fa34cf2